### PR TITLE
Misc Code Cleans(Updated)

### PR DESCRIPTION
**Pull request by InfiniteEchoes**
This PR is a updated version of #505 since there are too much conflicts to resolve with regard to the newest main branch.

This PR includes the following updates wrt several PRs:
- Removed StructUnit(#495 )
- Updated `ImplItemKind::to_coq` and `TopLevelItem::Comment::to_coq`
- With these updates, removed the `render` dependency for `top_level.rs` (#478 ) and all current occurences of `TopLevelItem::Code` & `coq::Expression::Code`.

In particular, `TopLevelItem::Comment` now supports multi-line comments. The comment style is fixed to
```coq
(* Single line comment regardless of the content *)
```
and
```coq
(*
Multi line
    comment
 regardless of the 
    content
*)
```

In my opinion the issues being mentioned can be closed after this PR.

### CODE REVIEW

File: lib/src/coq.rs [hunk 1]
```diff
@@ -18,8 +18,9 @@ pub(crate) struct TopLevel<'a> {
 pub(crate) enum TopLevelItem<'a> {
     /// the Code variant is for those constructions
     /// that are not yet represented by the types in this file
+    #[allow(dead_code)]
     Code(Doc<'a>),
```
Comment (clarus) [inline lib/src/coq.rs:22]: OK top that `Code` is not used anymore! We can also remove this constructor.

File: lib/src/coq.rs [hunk 2]
```diff
@@ -70,6 +71,7 @@ pub(crate) enum DefinitionKind<'a> {
 pub(crate) enum Expression<'a> {
     /// the Code variant is for those constructions
     /// that are not yet represented by the types in this file
+    #[allow(dead_code)]
     Code(Doc<'a>),
```
Comment (clarus) [inline lib/src/coq.rs:75]: Same, we can now remove this `Code` constructor.

Comment (InfiniteEchoes) [reply lib/src/coq.rs:75]: Removing `<'a>` for `Expression` would produce a massive chain effect, letting the compiler to let you delete almost all lifetime params for almost all structs being defined in this file. I wonder if this is ok for you?

Comment (clarus) [reply lib/src/coq.rs]: I think it is fine to do it in this pull request, and the opportunity to remove something that was making a lot of the code more complex.

Comment (InfiniteEchoes) [reply lib/src/coq.rs]: Ok. The newest PR already removed them all

**Review by clarus — APPROVED** [review]

**Pull request merged**